src/include/daos_srv/ad_mem.h

```diff
@@ -94,6 +94,36 @@ int ad_blob_close(struct ad_blob_handle bh);
 int ad_blob_destroy(struct ad_blob_handle bh);
 struct umem_store *ad_blob_hdl2store(struct ad_blob_handle bh);
 
+enum ad_blob_iter_flags {
+	AD_ITER_SORT_BY_GROUP_ADDR	= (1 << 0),
+	AD_ITER_SORT_BY_GROUP_WEIGHT	= (1 << 1),
+	AD_ITER_INCLUDE_UNINIT_GROUP	= (1 << 2),
+	/* iter stop after iterating all groups within cur arean */
+	AD_ITER_ONLY_CUR_ARENA		= (1 << 3),
+};
+
+struct ad_blob_iter_param {
+	uint32_t		ip_arena_index;
+	uint32_t		ip_group_index;
+	enum ad_blob_iter_flags ip_flags;
+};
+
+struct ad_blob_group_info {
+	/** unit size in bytes, e.g., 64 bytes, 128 bytes */
+	uint32_t	gi_unit;
+	/** number of units in this group */
+	int32_t		gi_unit_nr;
+	/** number of free units in this group */
+	uint32_t	gi_unit_free;
+};
+
+int ad_blob_iter_prep(struct ad_blob_iter_param *param);
+int ad_blob_iter_start(struct ad_blob_handle bh, struct ad_blob_iter_param *param);
+int ad_blob_iter_next(struct ad_blob_handle bh, struct ad_blob_iter_param *param);
+int ad_blob_iter_finish(struct ad_blob_handle bh, struct ad_blob_iter_param *param);
+int ad_blob_iter_fetch(struct ad_blob_handle bh, struct ad_blob_iter_param *param,
+		       struct ad_blob_group_info *info);
```

Comment:
I think we can build a few other APIs for arena?
When I try to debug the code, the information I'd like to know includes:
- total number of arenas
- number of arenas of each type.
- Free bits in the arena
- Number of groups in the arena
- Number of groups for each size

Reply:
Sure, thanks for feedback, be able to collect arena and groups information in one iteration cycle might be better.